solidity/test/token/TokenBridgeCctp.t.sol

```diff
@@ -196,17 +238,87 @@ contract TokenBridgeCctpV1Test is Test {
         _expectOffChainLookUpRevert(message);
         tbDestination.getOffchainVerifyInfo(message);
 
-        bytes32 nonceId = messageTransmitterDestination.hashSourceAndNonce(
-            tbDestination.hyperlaneDomainToCircleDomain(origin),
-            tokenMessengerDestination.nextNonce()
+        bytes memory cctpMessage = _encodeCctpMessage(
+            cctpNonce,
+            cctpOrigin,
+            ""
+        );
+        bytes memory attestation = bytes("");
+        bytes memory metadata = abi.encode(cctpMessage, attestation);
+
+        vm.expectCall(
+            address(messageTransmitterDestination),
+            abi.encodeWithSelector(
+                MockCircleMessageTransmitter.receiveMessage.selector,
+                cctpMessage,
+                attestation
+            )
+        );
+        assertEq(tbDestination.verify(metadata, message), true);
+    }
+
+    function test_verify_revertsWhen_invalidNonce() public {
+        Quote[] memory quote = tbOrigin.quoteTransferRemote(
+            destination,
+            user.addressToBytes32(),
+            amount
+        );
+
+        vm.startPrank(user);
+        tokenOrigin.approve(address(tbOrigin), quote[1].amount);
+
+        // invalid nonce := nextNonce + 1
+        uint64 cctpNonce = tokenMessengerOrigin.nextNonce() + 1;
+        tbOrigin.transferRemote{value: quote[0].amount}(
+            destination,
+            user.addressToBytes32(),
+            amount
+        );
+        uint256 nonce = mailboxDestination.inboundProcessedNonce();
+        bytes memory message = mailboxDestination.inboundMessages(nonce);
```

Comment:
We can just just pass 0 directly since we know we've only sent one message in this test.

Should probably add  a helper to get the latest sent message to `MockMailbox`.